src/mngt/ocf_mngt_cache.c

```diff
@@ -635,7 +635,7 @@ static void _ocf_mngt_attach_cache_device(ocf_pipeline_t pipeline,
 	if (context->volume_size < OCF_CACHE_SIZE_MIN) {
 		ocf_cache_log(cache, log_err, "ERROR: Cache cache size must "
 			"be at least %llu [MiB]\n", OCF_CACHE_SIZE_MIN / MiB);
-		OCF_PL_FINISH_RET(context->pipeline, -OCF_ERR_START_CACHE_FAIL);
+		OCF_PL_FINISH_RET(context->pipeline, -OCF_ERR_NO_MEM);
```

Comment:
Is this error related to the 'low memory' issue or is it related to the too small device?

Reply:
I agree with @micrakow, this is not memory related problem. Currently we don't have an error code addressing this particular issue, so you have two options:
a) Act as if one of parameters (cache device name) was invalid, because it represents device that doesn't satisfy our requirements, thus return not so very informative `-OCF_ERR_INVAL`.
b) Add dedicated error code and return it.
The latter may be used in `casadm` to print more detailed error message and thus improve UX, so I'd opt for that one.

Reply:
Hope it's now fixed.